### PR TITLE
WebGPU: Fixed build settings for pthreads (close #583)

### LINKED ISSUE #583: WebGPU: fix build settings for pthreads

### PR DESCRIPTION
**Pull request by MikhailGorobets**

### CODE REVIEW

File: ThirdParty/dawn/CMakeLists.txt [hunk 1]
```diff
@@ -74,6 +74,49 @@ set(DAWN_ABSEIL_DIR         "${abseil-cpp_SOURCE_DIR}"     CACHE STRING "Directo
...
+        tint_lang_wgsl_writer_ast_printer
+        tint_lang_wgsl_writer_raise 
+        tint_lang_wgsl_writer_ir_to_program
+        tint_lang_wgsl_reader_program_to_ir
+        tint_utils_id
+        tint_utils_text
+        tint_utils_symbol
+        tint_utils_diagnostic
+        tint_utils_result
+    )
```
Comment (TheMostDiligent) [inline ThirdParty/dawn/CMakeLists.txt:111]: We have a helper CMake function [`find_targets_in_directory`](https://github.com/DiligentGraphics/DiligentCore/blob/master/BuildTools/CMake/BuildUtils.cmake#L207)

**Review by TheMostDiligent — CHANGES_REQUESTED** [review]
Looks like gtest also [requires these settings](https://github.com/DiligentGraphics/DiligentCore/actions/runs/10143483209/job/28044934942#step:5:1132).

**Review by TheMostDiligent — CHANGES_REQUESTED** [review]
Please add the `set_targets_emscripten_properties` function to CMake Utils that will set the required properties on Emscripten platform and no nothing on other:

```CMake
find_targets_in_directory(DAWN_TARGETS ${dawn_SOURCE_DIR})
set_targets_emscripten_properties (${DAWN_TARGETS})
```

**Comment by TheMostDiligent** [comment]
@MikhailGorobets 
Please create the server startup script file and copy it to the build directory so that the server can be easily started

**Comment by MikhailGorobets** [comment]
> @MikhailGorobets Please create the server startup script file and copy it to the build directory so that the server can be easily started

This clearly needs to be in the samples module. No one uses SimpleHTTPServer in production

**Pull request merged**